File: Python_bonus/Colleen/Colleen.py
#!/usr/bin/env python3

'''
    This program will print its own source when run.
'''

def print_source():
    source = '#!/usr/bin/env python3\n\n\'\'\'\n    This program will print its own source when run.\n\'\'\'\n\ndef print_source():\n    source = %r\n    print(source %% source)\n\ndef main():\n    \'\'\'\n        This function prints the source code\n    \'\'\'\n    print_source()\n\nif __name__ == "__main__":\n    main()'
    print(source % source)

def main():
    '''
        This function prints the source code
    '''
    print_source()

if __name__ == "__main__":
    main()

File: C/Colleen/Colleen.c
#include <stdio.h>

/*
    This program will print its own source when run.
*/

int print_source() {
    char *source = "#include <stdio.h>%c%c/*%c    This program will print its own source when run.%c*/%c%cint print_source() {%c    char *source = %c%s%c;%c    return (printf(source, 10, 10, 10, 10, 10, 10, 10, 34, source, 34, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10));%c}%c%cint main() {%c    /*%c        This function prints the source code%c    */%c    print_source();%c    return 0;%c}%c";
    return (printf(source, 10, 10, 10, 10, 10, 10, 10, 34, source, 34, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10));
}

int main() {
    /*
        This function prints the source code
    */
    print_source();
    return 0;
}
